tests/integration/ceph_mgr_test.go

```diff
@@ -0,0 +1,250 @@
+}
+
+func (suite *CephMgrSuite) TearDownSuite() {
+	suite.cluster.Teardown()
+}
+
+func (suite *CephMgrSuite) execute(command []string) (error, string) {
+	orchCommand := append([]string{"orch"}, command...)
+	return suite.cluster.installer.Execute("ceph", orchCommand, suite.namespace)
+}
+
+func (suite *CephMgrSuite) waitForOrchestrationModule() {
+	for timeout := 0; timeout < 30; timeout++ {
+		err, output := suite.execute([]string{"status"})
+		logger.Info("%s", output)
+		if err == nil {
+			logger.Info("Rook Toolbox ready to execute commands")
+			break
+		}
+		time.Sleep(2 * time.Second)
+	}
+}
+func (suite *CephMgrSuite) TestDeviceLs() {
+	logger.Info("Testing .... <ceph orch device ls>")
+	err, device_list := suite.execute([]string{"device", "ls"})
+	assert.Nil(suite.T(), err)
+	logger.Infof("output = %s", device_list)
+}
+
+func (suite *CephMgrSuite) TestStatus() {
+	logger.Info("Testing .... <ceph orch status>")
+	err, status := suite.execute([]string{"status"})
+	assert.Nil(suite.T(), err)
+	logger.Infof("output = %s", status)
+
+	assert.Equal(suite.T(), status, "Backend: rook\nAvailable: True")
+}
+
+func (suite *CephMgrSuite) TestHostLs() {
+	logger.Info("Testing .... <ceph orch host ls>")
+
+	// Get the orchestrator hosts
+	err, output := suite.execute([]string{"host", "ls", "json"})
+	assert.Nil(suite.T(), err)
+	logger.Infof("output = %s", output)
+
+	hosts := []byte(output)
+	var hostsList []host
+
+	err = json.Unmarshal(hosts, &hostsList)
+	if err != nil {
+		assert.Nil(suite.T(), err)
+	}
+
+	var hostOutput []string
+	for _, hostItem := range hostsList {
+		hostOutput = append(hostOutput, hostItem.Addr)
+	}
+	sort.Strings(hostOutput)
+
+	// get the k8s nodes
+	nodes, err := k8sutil.GetNodeHostNames(suite.k8sh.Clientset)
+	assert.Nil(suite.T(), err)
+
+	k8sNodes := make([]string, 0, len(nodes))
+	for k := range nodes {
+		k8sNodes = append(k8sNodes, k)
+	}
+	sort.Strings(k8sNodes)
+
+	// nodes and hosts must be the same
+	assert.Equal(suite.T(), hostOutput, k8sNodes)
+}
+
+func (suite *CephMgrSuite) TestCreateOSD() {
+	logger.Info("Testing .... <ceph orch create OSD>")
+
+	// Get the first available device
+	err, deviceList := suite.execute([]string{"device", "ls", "--format", "json"})
+	assert.Nil(suite.T(), err)
+	logger.Infof("output = %s", deviceList)
+
+	inventory := make([]map[string]interface{}, 0)
+
+	err = json.Unmarshal([]byte(deviceList), &inventory)
+	assert.Nil(suite.T(), err)
+
+	selectedNode := ""
+	selectedDevice := ""
+	for _, node := range inventory {
+		for _, device := range node["devices"].([]interface{}) {
+			if device.(map[string]interface{})["available"].(bool) {
+				selectedNode = node["name"].(string)
+				selectedDevice = strings.TrimPrefix(device.(map[string]interface{})["path"].(string), "/dev/")
+				break
+			}
+		}
+		if selectedDevice != "" {
+			break
+		}
+	}
+	assert.NotEqual(suite.T(), "", selectedDevice, "No devices available to create test OSD")
```

Comment:
@jmolmo The CI is showing OSDs couldn't be created:
```
--- FAIL: TestCephMgrSuite (184.34s)
    --- FAIL: TestCephMgrSuite/TestCreateOSD (1.54s)
        ceph_mgr_test.go:214: 
            	Error Trace:	ceph_mgr_test.go:214
            	Error:      	Should not be: ""
            	Test:       	TestCephMgrSuite/TestCreateOSD
            	Messages:   	No devices available to create test OSD
```
A device should be available if the device is wiped from a previous test with the `WipeClusterDisks()` method. I see that it is called in the test:
```
job.batch/rook-ceph-disk-wipe-3 created
```
But your test output doesn't show any devices detected:
```
 integrationTest: output = [{"addr": "ip-172-31-36-37", "devices": [], "labels": [], "name": "ip-172-31-36-37"}]
```

Reply:
We do not have any error calling the "ceph orch device ls" command, however it does not return any information.
This probably means the the discovery daemon is not running in each host, or if it is running is not generating information about local disks in the device configmap. 
Bad news because taking into account that we are running the test in AWS....  maybe we cannot get this information.... 
@travisn What do you thing?

Reply:
Ok I see that the discover daemon is running and discovering that the disks are not available. The problem is that `h.WipeClusterDisks(namespace)` is called *after* the operator is started. If you're going to rely on the discovery results, you'll need to move the call to wipe disks *before* the operator is started. For example, it should be the first thing that the `InstallRook()` method calls.

Reply:
Great. thanks for discovery!! :-).  Really happy if the problem is only this!!

Reply:
Wiping disk first needs to access to nodes, so we need to move the disk wiping after the RBAC definition to avoid problems like:
`installer: failed to wipe cluster disks. failed to get node hostnames. failed to get k8s nodes. nodes is forbidden: User "system:anonymous" cannot list resource "nodes" in API group "" at the cluster scope. trying again...`